sycl/include/CL/sycl/program.hpp

```diff
@@ -301,14 +301,14 @@ class __SYCL_EXPORT program {
   /// \return a specialization constant instance corresponding to given type ID
   ///         passed as a template parameter
   template <typename ID, typename T>
-  experimental::spec_constant<T, ID> set_spec_constant(T Cst) {
-    constexpr const char *Name = detail::SpecConstantInfo<ID>::getName();
+  experimental::spec_constant<T, ID> set_spec_constant(T __UNUSED_ON_SYCL_DEVICE(Cst)) {   
```

Comment:
I think it's worth considering an alternative approach, which seems to be more readable/easier to maintain.

New macro definitions are used for the functions which code is `ifdef`-ed already, so it makes sense to apply the pattern commonly used in LLVM codebase:
```c++
#ifdef/ifndef __SYCL_DEVICE_TYPE__
// use to suppress a warning
(void)arg;
#else
// real use of arg
...
#endif
```

Reply:
Ok